numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -280,13 +280,23 @@ addUfuncs(PyObject *dictionary) {
                     "inner on the last dimension and broadcast on the rest \n"
                     "     \"(i),(i)->()\" \n",
                     0, inner1d_signature);
+    /*
+     * yes, this should not happen, but I (MHvK) just spent an hour looking at
+     * segfaults because I screwed up something that seemed totally unrelated.
+     */
+    if (f == NULL) {
+        return -1;
+    }
     PyDict_SetItemString(dictionary, "inner1d", f);
```

Comment:
In principle we should be checking the return value of `PyDict_SetItemString` too, but I'm happy for you to declare that out of scope for this PR.